cloud/ec2_util.go

```diff
@@ -466,15 +464,15 @@ func ec2CreateFleetResponseContainsInstance(createFleetResponse *ec2aws.CreateFl
 	return true
 }
 
-func validateEc2DescribeInstancesOutput(describeInstancesResponse *ec2aws.DescribeInstancesOutput) error {
+func validateEc2DescribeInstancesOutput(describeInstancesResponse *ec2.DescribeInstancesOutput) error {
 	catcher := grip.NewBasicCatcher()
 	for _, reservation := range describeInstancesResponse.Reservations {
 		if len(reservation.Instances) == 0 {
 			catcher.New("reservation missing instance")
 		} else {
 			instance := reservation.Instances[0]
 			catcher.NewWhen(instance.InstanceId == nil, "instance missing instance ID")
-			catcher.NewWhen(instance.State == nil || instance.State.Name == nil || len(*instance.State.Name) == 0, "instance missing state name")
+			catcher.NewWhen(instance.State == nil || len(instance.State.Name) == 0, "instance missing state name")
```

Comment:
I would do `instance.State.Name == ""` for the comparison here, using the length can be confusing since that is usually associated with slice comparisons.